rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/interactor/CovidPatientHistoryActivityInteractor.java

```diff
@@ -70,7 +70,7 @@ private String getValues(Obs obs, Map<String, String> formWidgetKeyToTextMap) {
         List<String> values = new ArrayList<>();
         // for checkboxes, get keys from key-value map
         Map<String, Object> keyValPairs = obs.getKeyValPairs();
-        List<Object> obsValues = Utils.isEmptyMap(keyValPairs) ? obs.getValues()
+        List<Object> obsValues = keyValPairs == null ? obs.getValues() : !Utils.isEmptyMap(keyValPairs) ? obs.getValues()
```

Comment:
@syedowaisali90 It's not clear to me, could you maybe explain what this logic achieves?

Reply:
we need `obs.getValues()` if `keyValPairs` is not empty or `null` because `obs.getValues()` contains patient name, dob, etc.
I have noticed in previous logic so it's got `obs.getValues()` when `keyValPairs` is empty or null that's why the patient name and dob showing empty.

Reply:
Do you mean that the `obs` from the  the patient details radio buttons have keyValPairs as not empty?

Reply:
yes

Reply:
What is contained in them?

Reply:
Contained patient data, if I revert the changes so the patient detail showing empty due to the previous logic.

Reply:
The previous logic checks if the keyValPairs are empty and if they're not, it uses the keys in there. If it's the case that the keyValPairs are not empty in this case, I'm wondering why the previous logic is failing?

Reply:
Could you share json of what the fields look like in the event?

Reply:
https://github.com/onaio/rdt-standard/pull/634#discussion_r532537127 yes keyValPairs found "patient_name" -> "Covid patient 2" but when we call a method `getValue("patient_name", formWidgetKeyToTextMap)` it return empty because `formWidgetKeyToTextMap` has empty value with "patient_name"

Reply:
https://github.com/onaio/rdt-standard/pull/634#discussion_r532537303 event obs
```json
{
      "fieldType": "formsubmissionField",
      "fieldDataType": "text",
      "fieldCode": "lbl_patient_name",
      "parentCode": "",
      "values": [
        "Covid Patient 2"
      ],
      "set": [],
      "formSubmissionField": "lbl_patient_name",
      "humanReadableValues": [],
      "saveObsAsArray": false
}
```

Reply:
Do we know why the value is empty? Could you try debugging the `formWidgetKeyToTextMap` creation process to see why those keys are missed?

Reply:
I have debugged the code of `formWidgetKeyToTextMap` creation so the value is empty because we loaded the fields from JSON from so the form by default doesn't have any value that's why the patient_name has an empty value.
https://github.com/onaio/rdt-standard/blob/20cc980aa0e0114acf36c6f83e8ef5da90c69294/rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/FormKeyTextExtractionUtil.java#L25-L66

Reply:
Ok, can we add placeholder text for the options in that case?

Reply:
Cool, do you have a possible solution?

Reply:
with the current changes that I have committed with this PR is working fine so we can go with this now or you could suggest to me another solution.

Reply:
Could you break the ternary operation into an if-statement? I don't really follow what it's doing atm.